### ISSUE TITLE
On Windows, .md files do not have correct value for last-updated timestamp

### ISSUE THREAD

**Issue by markproxy** [akosbalasko/yarle#486]
I installed yarle for Windows and ran it with default settings on a .enex file. The .md files that were generated have the correct values for their creation time, but their modified/last-updated times are all set to the moment they were generated.

Looking at the [setFileDates()](https://github.com/akosbalasko/yarle/blob/a67d1c8c80e0a5ac6f851998647011b2b59e7261/src/utils/content-utils.ts#L117) function, it looks to me like it _should_ work. The call to fs.utimesSync() should synchronously set the modified and accessed times, and the subsequent call to utimes() should asynchronously set the creation time (and, since the mtime and atime parameters are not provided, it looks to me like the modified and accessed times should be left as-is). So I'm afraid I can't point to a likely source of trouble.

**Comment by akosbalasko** [comment]
Hi @markproxy ,
Thank you for raising this up, I'll investigate it. 
Cheers,
akos

**Comment by markproxy** [comment]
I just tried an export to Logseq on the same machine, and I see the same problem... but only for certain files, and so far I don't see a pattern. A lot of the files affected have internal links or are pointed to by internal links. Just throwing out an idea: are files that have internal links touched after their file timestamps are set?

**Comment by akosbalasko** [comment]
Hi @markproxy ,

nope, timestamps are not used in Yarle at all. Internal links are based on the name of the note. 
But one thing: which Evernote version do you use to export your notes ?  Because in the last couple of days many people reported that EN 10.58 exports the notes with data loss. maybe it is related, I still don't know.

**Comment by markproxy** [comment]
> Hi @markproxy ,
> 
> nope, timestamps are not used in Yarle at all

I think you might have misunderstood my meaning. I was suggesting that after Yarle makes it call to setFileDates(), it them modifies the .md file, thereby overwriting the modified timestamp of the .md file that it had just modified via setFileDates().

> But one thing: which Evernote version do you use to export your notes ? Because in the last couple of days many people reported that EN 10.58 exports the notes with data loss. maybe it is related, I still don't know.

My version info is as follows, but I've seen no indication of data loss, and the last-updated timestamps that I imported into UpNote are all correct. It seems to be an issue that specific to Yarle, and which applies to only certain files.

```
10.58.8-win-ddl-public (4175)
Editor: v172.19.3
Service: v1.69.5
```

**Comment by akosbalasko** [comment]
Hi @markproxy ,
oh, okay, I get it now, thank you for the clarification. 
Yes, thinking about it, Yarle has some post-process to handle inter-enex links and v10 tasks, these actions modify the content of the file, and hence the mod. date as well. Thank you for raising this, I didn't consider these aspects. 
I think now I can reproduce the problem as a test in order to fix it. 
Thanks again!

**Comment by bts** [comment]
FWIW I am also experiencing this on macOS for notes with internal links

**Comment by akosbalasko** [comment]
fixed and released in 5.5.7.
